### PR TITLE
#164046216 Users can comment on articles and other comments

### PR DESCRIPTION
**Pull request by cob04**
**Description**

This Pull request adds the ability for users to comment on articles and other comments. Authors can also edit or delete the comments they create.
These endpoints have been added:

**POST** `/api/articles/<slug>/comments/`       - Create an article comment 
**GET** `/api/articles/<slug>/comments/`          - List article comments

**GET** `/api/articles/<slug>/comments/<pk>/`        -  Get an article comment with nested comments
**PUT** `/api/articles/<slug>/comments/<pk>/`        - Update a comment 
**PATCH** `/api/articles/<slug>/comments/<pk>/`    - Update a comment
**DELETE** `/api/articles/<slug>/comments/<pk>/`        - Delete a comment
**POST** `/api/articles/<slug>/comments/<pk>/`        - Create a comment for a  another comment 

<hr>
**Type of change**

- [x] new feature(non-breaking change which adds functionality).

<hr>

**How has this been tested**

- [x] End to End
- [x] Integration
<hr>

**How to test this manually**
Using Postman (or any other HTTP client), navigate to [https://ah-legion-staging-pr-28.herokuapp.com](https://ah-legion-staging-pr-28.herokuapp.com) And use the endpoints listed above.

<hr>
**Checklist:**

- [x] This follows all the guidelines for this project.

**Related Stories**
<hr>

[#164046253](https://www.pivotaltracker.com/n/projects/2313280/stories/164046253)

### CODE REVIEW

**Review by MandelaK — APPROVED** [review]
Looks good

**Review by Etomovich — APPROVED** [review]

**Review by Etomovich — APPROVED** [review]

**Review by BrianSerem — APPROVED** [review]

File: requirements.txt [hunk 1]
```diff
@@ -62,3 +62,7 @@ uritemplate==3.0.0
 urllib3==1.24.1
 whitenoise==4.1.2
 wrapt==1.11.1
+sendgrid==5.6.0
+python-http-client==3.1.0
+factory-boy==2.11.1
+readtime==1.1.0
```
Comment (abulojoshua1) [inline requirements.txt]: @cob04, why does your PR contain these dependencies?

File: authors/wsgi.py [hunk 2]
```diff
@@ -11,6 +11,7 @@
 
 from django.core.wsgi import get_wsgi_application
 
+
```
Comment (abulojoshua1) [inline authors/wsgi.py]: @cob04, why are you making changes to the `wsgi.py` file, does it add value to your PT board story?

File: authors/settings/base.py [hunk 3]
```diff
@@ -161,9 +163,11 @@
 SOCIAL_AUTH_FACEBOOK_KEY = config('SOCIAL_AUTH_FACEBOOK_KEY')
 SOCIAL_AUTH_FACEBOOK_SECRET = config('SOCIAL_AUTH_FACEBOOK_SECRET')
 SOCIAL_AUTH_FACEBOOK_SCOPE = ['email']
+
 SOCIAL_AUTH_FACEBOOK_PROFILE_EXTRA_PARAMS = {
     'fields': 'id, name, email',
 }
+
```
Comment (abulojoshua1) [inline authors/settings/base.py]: @cob04, why do you have changes in this file, do the changes add value to your PR?

File: authors/apps/articles/models.py [hunk 4]
```diff
@@ -1,7 +1,14 @@
+import readtime
```
Comment (abulojoshua1) [inline authors/apps/articles/models.py:1]: @cob04, why are you importing `readtime` here?

File: authors/settings/base.py (hunk 3 above)
Comment (cob04) [reply authors/settings/base.py]: This is deliberate, the empty spaces make the settings more readable.

File: authors/apps/articles/models.py (hunk 4 above)
Comment (cob04) [reply authors/apps/articles/models.py:1]: let me clean it up and run tests.

File: authors/settings/base.py (hunk 3 above)
Comment (abulojoshua1) [reply authors/settings/base.py]: @cob04, it would be best to do that clean up in a new PR.

File: authors/apps/articles/models.py (hunk 4 above)
Comment (cob04) [reply authors/apps/articles/models.py:1]: actually readtime is necessary here,  tests are failing.

File: requirements.txt (hunk 1 above)
Comment (cob04) [reply requirements.txt]: factory-boy is my dependency for use in tests. the rest I probably picked up from rebasing.
let me clean that up.

File: authors/apps/articles/models.py (hunk 4 above)
Comment (abulojoshua1) [reply authors/apps/articles/models.py:1]: @cob04, do you mean that readtime as a dependency was not added in the test file?

File: authors/settings/base.py (hunk 3 above)
Comment (cob04) [reply authors/settings/base.py]: I get you, I'll reverse those changes.

File: authors/apps/articles/models.py (hunk 4 above)
Comment (cob04) [reply authors/apps/articles/models.py:1]: I did not add it, it was there already there in `models.py` but it got displaced when I added my dependency imports.

**Review by abulojoshua1 — APPROVED** [review]
LGTM

**Pull request merged**